File: dcp-solutions/2024-09-24/solution.go
```markdown
package main

import (
	"container/heap"
	"fmt"
	"math/rand"
	"strconv"
)

type RandomizedSet struct {
	nums    []int
	next    map[int]int
}

func (rs *RandomizedSet) insert(val int) bool {
	if rs.next[val] == 0 {
		rs.nums = append(rs.nums, val)
		rs.next[val] = len(rs.nums) - 1
		return true
	}
	return false
}

func (rs *RandomizedSet) remove(val int) bool {
	if rs.next[val] != 0 {
		last := rs.nums[len(rs.nums)-1]
		rs.nums[rs.next[val]] = last
		rs.next[last] = rs.next[val]
		rs.next[val] = 0

		idx := rs.search(last)
		rs.nums[idx] = val
		rs.next[val] = idx
		rs.nums = rs.nums[:idx+1]
		return true
	}
	return false
}

func (rs *RandomizedSet) getRandom() int {
	rand.Seed(0x1234567890abcdef)
	return rs.nums[rand.Intn(len(rs.nums))]
}

func (rs *RandomizedSet) search(val int) int {
	idx := rs.next[val]
	for idx != 0 && rs.nums[idx] != val {
		idx = rs.next[rs.nums[idx]]
	}
	return idx
}

func main() {}
```

File: dcp-solutions/2024-09-24/solution.js
```javascript
class RandomizedSet {
    constructor() {
        this.store = new Map();
    }

    add(val) {
        if (!this.store.has(val)) {
            this.store.set(val, [...this.store.size]);
            return true;
        }
        return false;
    }

    remove(val) {
        let idx = this.store.get(val);
        if (idx !== val) {
            let lastIdxVal = this.store.keys().next().value;
            [this.store.set(idx), this.store.set(lastIdxVal)] = [lastIdxVal, idx];
        }
        this.store.delete(val);
        return true;
    }

    getRandom() {
        return this.store.keys().next().value;
    }
}
```